File: leetcode/algorithms/queue/number-of-recent-calls.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Binary Search"
*/

class RecentCounter {
    private LinkedList<Integer> pingsQueue;
    public RecentCounter() {
        pingsQueue = new LinkedList<Integer>();
        
    }
    
    public int ping(int t) {
        pingsQueue.addLast(t);
        while(pingsQueue.size()>0 && t-pingsQueue.get(0)>3000){
            pingsQueue.removeFirst();
        }
        
        return pingsQueue.size();
    }
}

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter obj = new RecentCounter();
 * int param_1 = obj.ping(t);
 */

File: leetcode/algorithms/queue/number-of-recent-calls.js
/*
    leetcode
    category: "Algorithms"
    problem title: "Binary Search"
*/


var RecentCounter = function() {
    
    this._queue = [];
    
};

/** 
 * @param {number} t
 * @return {number}
 */
RecentCounter.prototype.ping = function(t) {
    let removals = 0,i=0;
    while(i<this._queue.length && t-this._queue[i]>3000){
        ++i;
    }
    this._queue.splice(0,i);
    this._queue.push(t);
    return this._queue.length;
};

/** 
 * Your RecentCounter object will be instantiated and called as such:
 * var obj = new RecentCounter()
 * var param_1 = obj.ping(t)
 */